comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -142,6 +199,27 @@ func (ac *AutoConfig) checkTagFreshness(ctx context.Context) {
 	}
 }
 
+// Start will listen to the service channels before anything is sent to them
+// Usually, Start and Stop methods should not be in the component interface as it should be handled using Lifecycle hooks.
+// We make exceptions here because we need to disable it at runtime.
+func (ac *AutoConfig) Start() {
+	listeners.RegisterListeners(ac.serviceListenerFactories)
+	providers.RegisterProviders(ac.providerCatalog)
+	// Where to look for check templates if no custom path is defined
+	config.Datadog.SetDefault("autoconf_template_dir", "/datadog/check_configs")
```

Comment:
Curious why this is duplicated here when this default [is set in the global config setup here](https://github.com/DataDog/datadog-agent/blob/5d68596816649140d4305aacf430dd803cd7748d/pkg/config/setup/config.go#L580)?

Can we remove this here, and also move `autoconf_template_url_timeout` to `pkg/config/setup/config.go`?

(Ideally we should not be accessing `config.Datadog` from within a component in favor of the config component - but moving this out of the component impl seems like an even better alternative)